### PR TITLE
adds new emag feature to the BSA 

### PR DESCRIPTION
**Pull request by IHateGeese**
## About The Pull Request

Adds a feature where if you emag the BSA console the next shot will re-direct the blast target to the person who fired.

## Why It's Good For The Game

Adds a funny quirky new emag feature and adds some new options for traitors to mess with the station goals.

## Changelog
:cl:
add: emagging the BSA explodes the next person to fire it.
/:cl:

### CODE REVIEW

**Comment by IHateGeese** [comment]
this PR was minorly helped by @Rhials go hype him up at rhials#3573

File: code/modules/station_goals/bsa.dm [hunk 1]
```diff
@@ -363,7 +363,10 @@ GLOBAL_VAR_INIT(bsa_unlock, FALSE)
 		notice = "Cannon unpowered!"
 		return
 	notice = null
-	cannon.fire(user, get_impact_turf())
+	var/turf/target_turf = get_impact_turf()
+	if(obj_flags & EMAGGED)
+		target_turf = get_turf(src)
```
Comment (Fikou) [inline code/modules/station_goals/bsa.dm:368]: shouldnt this behavior be in get_impact_turf ??

Comment (IHateGeese) [reply code/modules/station_goals/bsa.dm]: fixed

**Comment by DaydreamIQ** [comment]
Revenants on their way to make the station explode one emag at a time

**Comment by AviiCinders** [comment]
Unfathomably based do it

**Comment by Korusho** [comment]
Literally best PR of 20203

**Comment by RiverMoth** [comment]
HOLY SHIT THIS IS THE GREATEST PR I HAVE EVER SEEN!!!!!

**Comment by private-tristan** [comment]
Will the resulting explosion destroy the BSA?

**Comment by IHateGeese** [comment]
> 

yes it will completely destroy it

**Comment by Rocko1345** [comment]
Once again the Emag has become a tool for wonderful trolling

**Comment by AviiCinders** [comment]
WE DID IT BOYS LETS GO

**Pull request merged**